### PR TITLE
Add new packages: spdx-satisfies & spdx-correct

### PR DESCRIPTION
**Pull request by jinwoo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/spdx-correct/spdx-correct-tests.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import * as spdxCorrect from 'spdx-correct';
```
Comment (sandersn) [inline types/spdx-correct/spdx-correct-tests.ts]: please use `import spdxCorrect = require('spdx-correct')` to import commonjs modules, which use the `export = ` syntax.

File: types/spdx-correct/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for spdx-correct 2.0
+// Project: https://github.com/jslicense/spdx-correct.js#readme
+// Definitions by: Jinwoo Lee <https://github.com/jinwoo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function spdxCorrect(identifier: string): string|null;
+declare namespace spdxCorrect {
```
Comment (sandersn) [inline types/spdx-correct/index.d.ts]: The namespace-merge hack may stop working in the future as node starts adding module support. Instead just use the `import x = require('x')` syntax to import.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/spdx-correct/index.d.ts (hunk 2 above)
Comment (jinwoo) [reply types/spdx-correct/index.d.ts]: Oh, ok. I've been wondering how typescript will support ES6 modules in Node.

Done.

File: types/spdx-correct/spdx-correct-tests.ts (hunk 1 above)
Comment (jinwoo) [reply types/spdx-correct/spdx-correct-tests.ts]: Done

**Review by sandersn — APPROVED** [review]

**Pull request merged**